### PR TITLE
Core's Rust driver encryption configuration

### PR DESCRIPTION
**Pull request by lolski**
## Usage and product changes

Implement TLS encryption for the Core's Rust driver, which can be configured through the `ConnectionSettings` object.

## Implementation

- Propagate the TLS configuration in `ConnectionSettings` into GRPC channel

### CODE REVIEW

File: rust/src/connection/network/channel.rs [hunk 1]
```diff
@@ -51,7 +51,10 @@ pub(super) fn open_callcred_channel(
     credential: Credential,
     connection_settings: ConnectionSettings,
 ) -> Result<(CallCredChannel, Arc<CallCredentials>)> {
-    let mut builder = Channel::builder(address.into_uri()); // TODO: configure TLS
+    let mut builder = Channel::builder(address.into_uri());
+    if connection_settings.is_tls_enabled() {
+        builder = builder.tls_config(connection_settings.tls_config().clone().unwrap())?;
```
Comment (farost) [inline rust/src/connection/network/channel.rs:56]: Didn't you say that TLS can be enabled even without a config?

Comment (farost) [reply rust/src/connection/network/channel.rs:56]: But we `unwrap` the user-specified config here, which is not correct. What are we going to do if we pass `(true, None)`?

**Review by farost — CHANGES_REQUESTED** [review]
Didn't get it yet!

File: rust/src/connection/network/channel.rs (hunk 1 above)
Comment (lolski) [reply rust/src/connection/network/channel.rs]: The TLS config object is unset if TLS is disabled, and set if TLS is enabled regardless of whether the CA is set or not. I've changed the `unwrap` to `expect` so it is clarified better to readers: 7dead0549d984c0c1ca4650d5df579f7577f0820.

**Comment by lolski** [comment]
I've merged this PR since @farost is away, and that I believe the issue he raised has been addressed.

**Pull request merged**